Spec/RealtimeClient.swift

```diff
@@ -352,6 +352,478 @@ class RealtimeClient: QuickSpec {
+
+                            guard let errorMessage = errorMessages.first else {
+                                fail("Missing ERROR protocol message"); partialDone(); return
+                            }
+                            expect(errorMessage.channel).to(contain("test"))
+                            expect(errorMessage.error?.code).to(equal(error.code))
+                            partialDone()
+                        }
+
+                        let tokenParams = ARTTokenParams()
+                        tokenParams.capability = "{\"test\":[\"subscribe\"]}"
+
+                        client.auth.authorize(tokenParams, options: nil) { tokenDetails, error in
+                            expect(error).to(beNil())
+                            guard let tokenDetails = tokenDetails else {
+                                fail("TokenDetails is nil"); partialDone(); return
+                            }
+                            expect(tokenDetails.token).toNot(equal(testToken))
+                            expect(tokenDetails.capability).to(equal(tokenParams.capability))
+                            partialDone()
+                        }
+                    }
+
+                    expect(client.auth.tokenDetails?.token).toNot(equal(testToken))
+                }
+
+                // RTC8a2
+                it("when the authentication token change fails, client should receive an ERROR ProtocolMessage triggering the connection to transition to the FAILED state") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    options.clientId = "ios"
+                    options.useTokenAuth = true
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    client.setTransportClass(TestProxyTransport.self)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connected) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+                            done()
+                        }
+                        client.connect()
+                    }
+
+                    var connectionError: NSError?
+                    var authError: NSError?
+
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+
+                        client.connection.once(.Failed) { stateChange in
+                            guard let stateChange = stateChange else {
+                                fail("ConnectionStateChange is nil"); partialDone(); return
+                            }
+                            expect(stateChange.previous).to(equal(ARTRealtimeConnectionState.Connected))
+                            expect(stateChange.reason).toNot(beNil())
```

Comment:
Can we check for a particular error code or type here please so that we know we're testing the correct thing?